### PR TITLE
Migrate to Fluentd v0.14 API

### PR DESCRIPTION
**Pull request by okkez**
See also: http://docs.fluentd.org/v0.14/articles/plugin-update-from-v12

### CODE REVIEW

**Comment by okkez** [comment]
I've confirmed with tutorial in README.md.

**Comment by kazegusuri** [comment]
Thank you for a great contribution!
I really use new features of v0.14, but I am not sure that v0.12 support should be dropped. I still use Fluentd v0.12 in production and __doesn't__ use v0.14 yet because of plugin changes or stability or ...
This is a plugin for monitoring.  If there still exists old version of Fluentd, these kind of plugin should support , IMO. How do you think?

**Comment by okkez** [comment]
It is very difficult to support Fluentd v0.12.x after migrating to Fluentd v0.14 API.
I recommend that we can create new branch to keep maintaining this plugin for Fluentd v0.12.x.
We can manage this plugin's version number as 0.3.x is for Fluentd v0.12.x and 0.4.x or 1.x.x for Fluentd v0.14.x. This is same as [fluent-plugin-s3](https://github.com/fluent/fluent-plugin-s3).

**Comment by okkez** [comment]
Rebased.

File: README.md [hunk 1]
```diff
@@ -4,6 +4,15 @@
 
 A fluent plugin that instruments metrics from records and exposes them via web interface. Intended to be used together with a [Prometheus server](https://github.com/prometheus/prometheus).
 
+If you are using Fluentd v0.10, you have to explicitly install [fluent-plugin-record-reformer](https://github.com/sonots/fluent-plugin-record-reformer) together. With Fluentd v0.12 or v0.14, there is no additional dependency.
+
+## Requirements
+
+| fluent-plugin-prometheus | fluentd    | ruby   |
+|--------------------------|------------|--------|
+| >= 0.4.0                 | >= v0.14.8 | >= 2.1 |
```
Comment (repeatedly) [inline README.md]: It should be 1.x.y: http://docs.fluentd.org/v0.12/articles/plugin-development#plugin-versioning-policy

**Comment by kazegusuri** [comment]
Now I merged this into master. I still keep supporting this plugin for Fluentd v0.12 in *v0.12* branch.
Thank you for your great work.

**Pull request merged**